<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1543</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1761</width>
      <height>931</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Исследование алгоритмов построения</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>180</y>
        <width>301</width>
        <height>351</height>
       </rect>
      </property>
      <property name="title">
       <string>Алгоритм рисования</string>
      </property>
      <widget class="QRadioButton" name="InternalButton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>221</width>
         <height>26</height>
        </rect>
       </property>
       <property name="text">
        <string> Библиотечная функция</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QRadioButton" name="VuButton">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>282</y>
         <width>17</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>280</y>
         <width>88</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Алгоритм Ву&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="BrWithoutStairsButton">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>232</y>
         <width>17</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>35</x>
         <y>221</y>
         <width>194</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Алгоритм Брезенхема с &lt;br/&gt;устранением ступенчатости&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>37</x>
         <y>171</y>
         <width>192</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Алгоритм Брезенхема с &lt;br/&gt;целочисленными данными&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="BrIntButton">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>182</y>
         <width>17</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QRadioButton" name="BrFloatButton">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>132</y>
         <width>17</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>35</x>
         <y>121</y>
         <width>194</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Алгоритм Брезенхема с &lt;br/&gt;действительными данными&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>35</x>
         <y>71</y>
         <width>239</width>
         <height>40</height>
        </rect>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Алгоритм цифрового &lt;br/&gt;дифференциального анализатора&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="DDAButton">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>82</y>
         <width>17</width>
         <height>18</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">buttonGroup</string>
       </attribute>
      </widget>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>500</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Построить</string>
      </property>
     </widget>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>30</y>
        <width>1021</width>
        <height>831</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="x1">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>30</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="y1">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>30</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="y1_2">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>70</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="x1_2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>70</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>111</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Начало отрезка:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Конец отрезка</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>0</y>
        <width>16</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>x</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>0</y>
        <width>16</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>y</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>540</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Построить спектр</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>580</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Очистить плоскость</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>180</y>
        <width>63</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Линия</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>270</y>
        <width>63</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Фон</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>20</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Текущий цвет линии</string>
      </property>
     </widget>
     <widget class="QPushButton" name="colorButton">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>50</y>
        <width>21</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QComboBox" name="lineColor">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>210</y>
        <width>141</width>
        <height>28</height>
       </rect>
      </property>
      <property name="currentText">
       <string>Белый</string>
      </property>
      <item>
       <property name="text">
        <string>Красный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Жёлтый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Оранжевый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Фиолетовый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Зелёный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Синий</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Чёрный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Белый</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="lineColor_2">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>300</y>
        <width>141</width>
        <height>28</height>
       </rect>
      </property>
      <property name="currentText">
       <string>Белый</string>
      </property>
      <item>
       <property name="text">
        <string>Красный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Жёлтый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Оранжевый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Фиолетовый</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Зелёный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Синий</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Чёрный</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Белый</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>241</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Шаг угла для спектра (в градусах)</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="spectreAngle">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>130</y>
        <width>41</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>640</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Замеры скорости работы</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>680</y>
        <width>431</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Исследование зависимости кол-ва ступенек от угла наклона</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_6">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>720</y>
        <width>461</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>7</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Исследование зависимости максимальной длины ступеньки от угла наклона</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>380</y>
        <width>171</width>
        <height>101</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>350</y>
        <width>171</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Информация о прямой</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1543</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
